apistar/document.py

```diff
@@ -136,9 +136,10 @@ def __init__(self,
 
         # Add in path fields for any "{param}" items that don't already have
         # a corresponding path field.
-        for path_name in url_path_names:
-            if path_name not in [field.name for field in path_fields]:
-                fields += [Field(name=path_name, location='path', required=True)]
+        field_names = [field.name for field in path_fields]
+        fields.extend([Field(name=path_name, location='path', required=True)
```

Comment:
There's no need to generate a list here so you can drop the `[`s and get rid of an allocation.

Reply:
Slightly slower since extending two lists is simply replacing a pointer, but who cares. It's clearer :)